### PR TITLE
feat(lib): Add support for the built-in Terraform data resource

### LINKED ISSUE #3142: lib: Support `terraform_data` built-in resource
### Description

Terraform 1.4 added support for a built-in `terraform_data` resource which can be used instead of the `null` resource. We should support it, too!

Note: We should ensure `convert` supports this as well!

### References

Terraform PR: https://github.com/hashicorp/terraform/pull/31757
Terraform Docs: https://developer.hashicorp.com/terraform/language/resources/terraform-data
Discuss thread asking for CDKTF support: https://discuss.hashicorp.com/t/how-to-use-terraform-data-in-cdk-tf/58390

### Help Wanted

- [X] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by ansgarm**
The code is based on the output of our provider-generation as that requires only a simple (import related) special handling in convert

Closes #3142

## Todos

- [x] Add convert support for Terraform Data Resource (requires adjusting the import)
- [x] Docs?

### CODE REVIEW

File: packages/@cdktf/hcl2cdk/test/__snapshots__/resources.test.ts.snap [hunk 1]
```diff
@@ -73,6 +73,27 @@ class MyConvertedCode extends TerraformStack {
 "
 `;
 
+exports[`resources built-in data resource snapshot typescript 1`] = `
+"import { Construct } from "constructs";
+import { TerraformVariable, DataResource, TerraformStack } from "cdktf";
+class MyConvertedCode extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+    /*The following providers are missing schema information and might need manual adjustments to synthesize correctly: terraform.
```
[the remark this answers is not in the document]
Comment (ansgarm) [reply packages/@cdktf/hcl2cdk/test/__snapshots__/resources.test.ts.snap:82]: Oh, good catch 😄

File: packages/cdktf/test/terraform-data-resource.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { ref } from "../lib/tfExpression";
+import { DataResource } from "../lib/terraform-data-resource";
+
+test("built-in Terraform data resource", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  new DataResource(stack, "test-data", {
+    input: ref("var.input"),
+    triggersReplace: [ref("var.triggersReplace1"), ref("var.triggersReplace2")],
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply packages/cdktf/test/terraform-data-resource.test.ts:13]: Yeah that makes sense to me 👍

File: website/docs/cdktf/concepts/resources.mdx [hunk 3]
```diff
@@ -1165,6 +1165,10 @@ securitygroup.NewSecurityGroup(stack, jsii.String("security2"), &securitygroup.S
 
 ## Special Cases
 
+### Built-in `terraform_data` resource
+
+There is a special resource built into Terraform itself. It is called [`terraform_data`](/terraform/language/resources/terraform-data) and implements the standard resource lifecycle, but does not directly take any other actions. In CDKTF it is exposed as the `TerraformData` class and you can import it directly from the `cdktf` package.
```
Comment (trujillo-adam) [inline website/docs/cdktf/concepts/resources.mdx:1170]:
```suggestion
The [`terraform_data`](/terraform/language/resources/terraform-data) resource implements the standard resource lifecycle but does not directly perform any other actions. In CDKTF, the resource is exposed as the `TerraformData` class and you can import it directly from the `cdktf` package.
```
I think we can be a little more concise. I also fixed some comma errors.

**Review by trujillo-adam — APPROVED** [review]
Made a few suggestions for the docs - pre-approving so you're not blocked from the edu side.

File: website/docs/cdktf/concepts/resources.mdx (hunk 3 above)
Comment (ansgarm) [reply website/docs/cdktf/concepts/resources.mdx]: Thank you so much @trujillo-adam 💯

**Pull request merged**